### ISSUE TITLE
Opening the tab stash tab in another window doesn't unstash in correct window

### ISSUE THREAD

**Issue by EmmanuelMess** [josh-berry/tab-stash#197]
### System Details

- **OS Version:** Ubuntu 20.04
- **Browser Version:** Waterfow G4.0.5.1 (64-bit)
- **Tab Stash Version:** 2.10.1

**Installed Browser Extensions:**

### Describe the Bug

1. Move stash tab to new window.
2. Open stashed tabs without deleting stash.
3. Tabs open in wrong window

### Expected Behavior

Tabs should open in a new window.

### Impact

Medium

<hr>
<em>Readers: If you are also experiencing this bug, please vote for it by giving the ORIGINAL POST a thumbs-up using the :smiley: button.  You are welcome to leave comments and discuss the issue if you have more details to add, but "Me too!" comments are not counted by the voting system.</em>

**Comment by josh-berry** [comment]
Let me see if I understand correctly:
1. You open the Tab Stash UI in a tab
2. You move that tab to its own/a different window
3. The Tab Stash UI still shows/manages tabs from the first window (where the Tab Stash UI was originally opened), but you were expecting the UI to immediately switch to managing the new window instead.

Assuming I got that right, this is actually by design. However, you can work around it by simply reloading the Tab Stash UI tab to force it to manage the new window.

It works this way so that it's easier for you to organize a window with a ton of tabs that should go to many different groups. For example, suppose you have 50 groups in Tab Stash, and 50 open tabs that need to go into both new and old groups (near both the top and bottom of your group list). In this situation, you can move the Tab Stash UI tab to its own window, sort through the tabs in the first window, and still take advantage of Tab Stash's multi-column view to move the tabs into the right groups. Doing this with the sidebar can be quite painful (lots more scrolling required).

Let me know if I understood you correctly, and if that explains what you're seeing or if you have any further questions!

**Comment by EmmanuelMess** [comment]
> 3. The Tab Stash UI still shows/manages tabs from the first window (where the Tab Stash UI was originally opened), but you were expecting the UI to immediately switch to managing the new window instead.

That's not it.

I expect opening a stash (though the "open all and keep stash" button) to open it in the current window, not the old one.

**Comment by josh-berry** [comment]
OK, then I am a bit confused--because you said, "That's not it." but then you also said, "I expect opening a stash (though the "open all and keep stash" button) to open it in the current window, not the old one.", which I think is the same thing I said in point 3.

What am I missing?

**Comment by EmmanuelMess** [comment]
So it is normal behaviour that the stashed tabs do not open in the same window as the "Tab Stash" tab?

**Comment by josh-berry** [comment]
Yes. Tab Stash remembers which window it was opened in and will continue to use that window.

If you want to switch it to a new window, just reload the UI (refresh the Tab Stash page or close and open the sidebar again), and it will use the new window.

**Comment by EmmanuelMess** [comment]
Can you tell the user that? as a UI notification?

**Comment by josh-berry** [comment]
I'll give it some thought, but I'm not sure how to do this yet in a way that's unobtrusive and doesn't bother the user after the first time (e.g. if someone prefers to do it regularly).
